protobuf/src/main/java/com/linecorp/armeria/server/protobuf/ProtobufRequestConverterFunctionProvider.java

```diff
@@ -57,23 +57,35 @@ static ResultType toResultType(Type type) {
             final ParameterizedType parameterizedType = (ParameterizedType) type;
             final Class<?> rawType = (Class<?>) parameterizedType.getRawType();
             if (List.class.isAssignableFrom(rawType)) {
-                final Class<?> typeArgument = (Class<?>) parameterizedType.getActualTypeArguments()[0];
-                if (isProtobufMessage(typeArgument)) {
+                final Type actualTypeArgument = parameterizedType.getActualTypeArguments()[0];
+                if (!(actualTypeArgument instanceof Class<?>)) {
+                    return ResultType.UNKNOWN;
+                }
+                if (isProtobufMessage((Class<?>) actualTypeArgument)) {
                     return ResultType.LIST_PROTOBUF;
                 }
             } else if (Set.class.isAssignableFrom(rawType)) {
-                final Class<?> typeArgument = (Class<?>) parameterizedType.getActualTypeArguments()[0];
-                if (isProtobufMessage(typeArgument)) {
+                final Type actualTypeArgument = parameterizedType.getActualTypeArguments()[0];
+                if (!(actualTypeArgument instanceof Class<?>)) {
+                    return ResultType.UNKNOWN;
+                }
+                if (isProtobufMessage((Class<?>) actualTypeArgument)) {
                     return ResultType.SET_PROTOBUF;
                 }
             } else if (Map.class.isAssignableFrom(rawType)) {
                 final Type[] typeArguments = parameterizedType.getActualTypeArguments();
+                if (!(typeArguments[0] instanceof Class<?>)) {
+                    return ResultType.UNKNOWN;
+                }
                 final Class<?> keyType = (Class<?>) typeArguments[0];
                 if (!String.class.isAssignableFrom(keyType)) {
                     throw new IllegalStateException(
```

Comment:
Shouldn't we return `return ResultType.UNKNOWN;` instead of raising an exception?